scripts/view-test-artifacts.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * Usage: `yarn view-test-artifacts <buildBatchId>`
+ *
+ * N.B. It is important to have your local environment configured for the correct codebuild account before running the script.
+ * This script caches resources, but in the case the local resource state is out of sync, you may need to wipe the asset directory
+ * that is printed when the script begins.
+ */
+
+import * as process from 'process';
+import * as path from 'path';
+import * as os from 'os';
+import * as fs from 'fs-extra';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { CodeBuild, S3 } from 'aws-sdk';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { SingleBar, Presets } from 'cli-progress';
+import * as execa from 'execa';
+
+const s3 = new S3();
+const progressBar = new SingleBar({}, Presets.shades_classic);
+
+type BuildStatus = 'FAILED' | 'FAULT' | 'IN_PROGRESS' | 'STOPPED' | 'SUCCEEDED' | 'TIMED_OUT';
+
+const buildStatusToIcon: Record<BuildStatus, string> = {
+  FAILED: '❌',
+  FAULT: '🚫',
+  IN_PROGRESS: '🏃',
+  STOPPED: '➖',
+  SUCCEEDED: '✅',
+  TIMED_OUT: '⏰',
+};
+
+/**
+ * Type modelling info we need to retrieve and display test artifacts in the UI.
+ */
+type TestArtifact = {
+  jobName: string;
+  buildStatus: BuildStatus;
+  artifactLocation?: string;
+};
+
+/**
+ * Utility method to filter out incomplete test artifact definitions.
+ * @param artifact the potentially incomplete artifact to validate
+ * @returns whether the artifact is complete or not
+ */
+const testArtifactIsComplete = (artifact: Partial<TestArtifact>): artifact is TestArtifact => artifact.jobName !== undefined;
```

Comment:
Shouldn't this be based on `buildStatus` not `jobName`?

Reply:
This check is intended towards filtering the tests without an identifier since Codebuild SDK call for getting batches could return `undefined` job_names. We have subsequent filtering for build status in place, so the in_progress ones are shown as such on the UI.
![Screenshot 2023-07-31 at 12 55 44 PM](https://github.com/aws-amplify/amplify-codegen/assets/55896475/1f5cacb7-1007-4156-a6f0-0d36ab6b37ec)